stan/math/opencl/kernels/check_symmetric.hpp

```diff
@@ -43,7 +44,7 @@ static const char *is_symmetric_kernel_code = STRINGIFY(
 /**
  * See the docs for \link kernels/check_symmetric.hpp check_symmetric() \endlink
  */
-const global_range_kernel<cl::Buffer, cl::Buffer, int, int, const double>
+const kernel_cl<read_buffer, write_buffer, int, int, const double>
```

Comment:
So this should be a `read_write_buffer`. @SteveBronder could you change this to read_write. I wasnt following this PR as closely and dont want to break anything.

Reply:
Since `write_buffers` wait for both their read and write queues to finish I think it's fine as is